### ISSUE TITLE
There's no easy way to see which version of Frontend a service is using

### ISSUE THREAD

**Issue by timpaul** [alphagov/govuk-frontend#1734]
If we added a comment or data attribute to our html or css, then it might make it easier to scour the web (and service subdomains) looking for which versions of GOV.UK Frontend people were using. This would help us track adoption and identify any blockers to upgrading.

**Comment by nickcolley** [comment]
Makes sense to try to add this to more places for redundancy incase people dont use the project in a way we expect.

Comments are commonly minified in production so perhaps something like:

## JavaScript

```javascript
window.GOVUKFrontend.version = 2.5.0
```

## CSS
```css
.govuk-frontend-version::after { content: '2.5.0' }
```

## HTML
```html
<html>
  <head>
    <meta name="govuk-frontend" content="2.5.0">
  </head>
</html>
```

**Comment by nickcolley** [comment]
One way you could incentive this is to make a community dashboard type thing, they get metrics on how performant their service is if they add certain tags to their page, then we can see what version they're using etc.

**Comment by 36degrees** [comment]
For CSS, could we set a variable on the `:root`?
```css
:root {
  --govuk-frontend-version: 3.5.0;
}
```

**Comment by nickcolley** [comment]
@36degrees that's nicer 👏

**Comment by nickcolley** [comment]
Interestingly https://www.wappalyzer.com/ can detect GOV.UK Frontend on the Design System website, my guess is by looking at the name of the UMD bundle which not every website will do.

**Comment by nickcolley** [comment]
Ah Matt H added it here: https://github.com/AliasIO/wappalyzer/pull/2939

**Comment by nickcolley** [comment]
> The HTTPArchive use Wappalyzer to categorise a lot of tech on the web on their monthly data collection. So in the future it should be possible to identify FE much more easily in their data.

**Comment by Nooshu** [comment]
Conversation being had on the [HTTPArchive discussion forum](https://discuss.httparchive.org/t/identifying-the-gov-uk-design-system-usage-across-the-web/1861) about identifying FE / DS.

**Comment by nickcolley** [comment]
Good to think about this scenario:

> In the accordion method you could have a scenario where they’ve only built the CSS for the accordion, exported a custom build of the JS just for the accordion.

**Comment by nickcolley** [comment]
Looks like we can combine different checks and assign them confidence ratios: https://github.com/AliasIO/wappalyzer/blob/master/src/apps.json#L7691-L7710

https://docs.wappalyzer.com/dev/specification.html#tags

Looks like we can use 'excludes' as well.

**Comment by nickcolley** [comment]
As a note, that tool doesnt support looking in CSS so perhaps it's not worth doing CSS version numbers?

**Comment by nickcolley** [comment]
Also looks like the UIKIt regex is incorrectly picking up our CSS? https://github.com/AliasIO/wappalyzer/blob/master/src/apps.json#L11335-L11343

**Comment by 36degrees** [comment]
It might also be worth making it possible to tell whether a user is `dist` or `package`.

**Comment by querkmachine** [comment]
Moving to blocked for the time being as we want to try some alternative approaches.

**Comment by stevenjmesser** [comment]
> Looks like we can combine different checks and assign them confidence ratios: https://github.com/AliasIO/wappalyzer/blob/master/src/apps.json#L7691-L7710
> 
> https://docs.wappalyzer.com/dev/specification.html#tags
> 
> Looks like we could in theory setup checks for the legacy frameworks and use 'excludes' as well.

Ah, this is useful, thanks! I've only set up one list so far (which produced some [results](https://docs.google.com/spreadsheets/d/1H7REK8lwQv7r6hB9P2ZcojtfEozcLXoIsS8kGqa551Q/edit?usp=sharing)) but it sounds like using multiple lists that make use of excludes could be another thing to try.
